code/modules/jobs/job_types/_job.dm

```diff
@@ -133,6 +155,112 @@
 	if(lock_flags || gimmick)
 		SSjob.job_manager_blacklisted |= title
 
+/// Does this job have any space to join?
+/datum/job/proc/has_space()
+	// How many slots does our group have?
+	var/group_slots = get_spawn_position_count(TRUE)
+	// Always available
+	if (group_slots == -1)
+		return TRUE
+	// If the department itself is saturated, return false
+	if (dynamic_spawn_group_multiplier && get_group_position_count() >= group_slots)
+		return FALSE
+	// If the role itself does not care about saturation, return true
+	if (total_positions == -1 || dynamic_spawn_group)
+		return TRUE
+	// If the role itself is saturated, return false
+	if (current_positions >= total_positions  + total_position_delta)
+		return FALSE
+	// Return true otherwise
+	return TRUE
+
+/// Calculates the number of positions currently present in a group
+/// Returns the number of players who are inside a role if the job hasn't reached
+/// its population limit, otherwise groups together roles by checking for their
+/// min_pop_redirect proxy role.
+/datum/job/proc/get_group_position_count()
+	var/spawn_group_size = current_positions * dynamic_spawn_group_multiplier
+	// Find all jobs that proxy to the target's spawn group
+	// This will mean that medical will count all of the players in medical
+	for (var/datum/job/group_job in SSjob.occupations)
+		// We already counted ourselves
+		if (group_job == src)
+			continue
+		// If this job proxies to us, then their proxy needs to be active
+		if (group_job.min_pop_redirect == type && SSjob.initial_players_to_assign < group_job.min_pop)
```

Comment:
```suggestion
		if (group_job.min_pop_redirect == src.type && SSjob.initial_players_to_assign < group_job.min_pop)
```
We have `group_job.type` in the code. Writing src will make it sure.

Reply:
No changes made: Unprefered style choice